src/IO/ADIOS/ADIOS2IOHandler.cpp

```diff
@@ -805,8 +805,15 @@ namespace detail
         ( std::is_same< T, rep >::value )
```

Comment:
Just to clarify my mind, when is this true?

Reply:
The ADIOS2 backend uses `unsigned char` for booleans, so it is true when trying to read any attribute of that type.

Reply:
I just checked, we also use `unsigned char` in ADIOS1: https://github.com/openPMD/openPMD-api/blob/0.12.0-alpha/src/IO/ADIOS/CommonADIOS1IOHandler.cpp#L48-L49